docs/design-survey-conditional-logic.md

```diff
@@ -0,0 +1,691 @@
+| [Matrix Dropdown](/Documentation/Library?id=questionmatrixdropdownmodel) | `{questionname.rowname.columnname}` |
+
+[View the "Complex Questions in Expressions" example](/Examples/Library?id=condition-complexquestions)
+
+In dynamic questions, use a zero-based index to access a specific value (question or cell):
+
+| Question Type | Syntax |
+| ------------- | ------ |
+| [Panel Dynamic](/Documentation/Library?id=questionpaneldynamicmodel) | `{dynamicpanelname[index].questionname}` |
+| [Matrix Dynamic](/Documentation/Library?id=questionmatrixdynamicmodel) | `{dynamicmatrixname[rowindex].columnname}` |
+
+In addition, Matrix questions support a `row` prefix that allows you to access cells on the same row: `{row.columnname}`. Similarly, you can use a `panel` prefix to access questions in the same Panel: `{panel.questionName}`.
+
+[View the "Conditions in Dynamic Questions" example](/Examples/Library?id=condition-dynamic)
+
+### Variables
+
+Variables are used to store JavaScript-calculated values. To create or change a variable, call the Survey's [`setVariable(name, value)`](https://surveyjs.io/Documentation/Library?id=surveymodel#setVariable) method. In the following code, this method sets a `currentYear` variable used to display the current year in an Html question:
+
+```js
+import { Model } from "survey-core";
+
+const surveyJson = {
+  "elements": [{
+    "name": "footer",
+    "type": "html",
+    "html": "&copy; 2015-{currentyear} Devsoft Baltic OÜ"
+  }]
+};
+
+const survey = new Model(surveyJson);
+
+survey.setVariable("currentyear", new Date().getFullYear());
+```
+
+If you need to get a variable's value, call the [`getVariable(name)`](https://surveyjs.io/Documentation/Library?id=surveymodel#getVariable) method. For example, the following code outputs the `currentyear` variable's value into the browser's console:
+
+```js
+console.log(survey.getVariable("currentyear"));
+```
+
+You can also call the [`getVariableNames()`](https://surveyjs.io/Documentation/Library?id=surveymodel#getVariableNames) method to get a list of all available variables:
+
+```js
+console.log(survey.getVariableNames()); // Outputs [ "currentyear" ]
+```
+
+### Calculated Values
+
+Calculated values allow you to register an [expression](#expressions) under a required name. If the expression includes [questions](#question-values), [variables](#variables), or [functions](#built-in-functions), it is recalculated each time their values are changed.
+
+To configure a calculated value, define the [`calculatedValues`](https://surveyjs.io/Documentation/Library?id=surveymodel#calculatedValues) array in the survey JSON definition. Each object in this array should contain the following fields:
+
+- `name` - A name that identifies the calculated value.
+- `expression` - An expression that returns the calculated value.
+- `includeIntoResult` - A Boolean property that specifies whether to include the calculated value into survey results.
```

Comment:
`includeIntoResult` - A Boolean property that specifies whether to include the calculated value in survey results.